src/features/auth/context.js

```diff
@@ -0,0 +1,176 @@
+import React, { createContext, useState, useCallback, useEffect } from 'react';
+
+import { useGuestLocale } from 'features/app/hooks/guestLocale';
+import AuthService from './services/AuthService';
+import { applyInterceptors, clearInterceptors } from './services/middleware';
+
+export const AuthContext = createContext();
+
+const defaultState = {
+  user: null,
+  session: null,
+  isLoading: false,
+};
+
+const initialState = () => {
+  const storedSession = JSON.parse(localStorage.getItem('session'));
+  if (storedSession) {
+    return { ...defaultState, session: storedSession, isLoading: true };
+  }
+  return defaultState;
+};
+
+// eslint-disable-next-line react/prop-types
+export const AuthProvider = ({ children, prepopulatedState, httpClient }) => {
+  const [state, setState] = useState(prepopulatedState || initialState);
+  const { user, session, isLoading } = state;
+
+  const { guestLocale } = useGuestLocale();
+
+  const validateSession = useCallback(async () => {
+    if (isLoading && session) {
+      try {
+        const response = await AuthService.validateToken();
+        const {
+          headers: { accessToken, client, uid },
+          data,
+        } = response;
+
+        setState({
+          session: { accessToken, client, uid },
+          user: data.user,
+          isLoading: false,
+        });
+      } catch {
+        setState(defaultState);
+      }
+    }
+  }, [isLoading, session]);
+
+  const signIn = useCallback(async (values) => {
+    const response = await AuthService.signIn(values);
+    const {
+      headers: { accessToken, client, uid },
+      data,
+    } = response;
+
+    setState({
+      session: { accessToken, client, uid },
+      user: data.user,
+      isLoading: false,
+    });
+
+    return response;
+  }, []);
+
+  const signUp = useCallback(async (values) => {
+    const response = await AuthService.signUp(values);
+    const {
+      headers: { accessToken, client, uid },
+      data,
+    } = response;
+
+    setState({
+      session: { accessToken, client, uid },
+      user: data.user,
+      isLoading: false,
+    });
+
+    return response;
+  }, []);
+
+  const updateUser = useCallback(async (values, passwordCheck = null) => {
+    const response = await AuthService.updateUser(values, passwordCheck);
+    const { data } = response;
```

Comment:
```suggestion
   const { data }  = await AuthService.updateUser(values, passwordCheck);
```

Reply:
if we do in that way, what we are going to return in the callback?

Reply:
Yeah I missed that, in some of them we return the response 👌